javascript/2-21/scripts/main.js

```diff
@@ -148,63 +148,87 @@ var imageList = [
+    if (event.target === $lightbox[0] || event.target === $('.close-button')[0]) {
+        $lightbox.toggleClass('inactive');
     }
 });
 
-lightbox.querySelector('.left-arrow').addEventListener('click', function(event) {
+$('.left-arrow').click(function(event) {
     event.preventDefault();
-    var currentImage = lightbox.querySelector('img');
+    var $currentImage = $('#lightbox img');
     var nextImage;
-    if (currentImage.getAttribute('data-item-id') === '0') {
+    if ($currentImage.attr('data-item-id') === '0') {
         nextImage = imageList[(imageList.length-1)];
     } else {
-        nextImage = imageList[Number(currentImage.getAttribute('data-item-id'))-1];
+        nextImage = imageList[Number($currentImage.attr('data-item-id'))-1];
     }
-    currentImage.setAttribute('src', nextImage.src);
-    currentImage.setAttribute('alt', nextImage.alt);
-    currentImage.setAttribute('data-item-id', nextImage.id);
+    $currentImage.attr('src', nextImage.src);
+    $currentImage.attr('alt', nextImage.alt);
+    $currentImage.attr('data-item-id', nextImage.id);
 });
 
-lightbox.querySelector('.right-arrow').addEventListener('click', function(event) {
+$('.right-arrow').click(function(event) {
     event.preventDefault();
-    var currentImage = lightbox.querySelector('img');
+    var $currentImage = $('#lightbox img');
     var nextImage;
-    if (Number(currentImage.getAttribute('data-item-id')) === (imageList.length-1)) {
+    if (Number($currentImage.attr('data-item-id')) === (imageList.length-1)) {
         nextImage = imageList[0];
     } else {
-        nextImage = imageList[Number(currentImage.getAttribute('data-item-id'))+1];
+        nextImage = imageList[Number($currentImage.attr('data-item-id'))+1];
     }
-    currentImage.setAttribute('src', nextImage.src);
-    currentImage.setAttribute('alt', nextImage.alt);
-    currentImage.setAttribute('data-item-id', nextImage.id);
+    $currentImage.attr('src', nextImage.src);
+    $currentImage.attr('alt', nextImage.alt);
+    $currentImage.attr('data-item-id', nextImage.id);
 });
 
 imageList.forEach(function(image, i) {
-        var preview = document.createElement('img');
-        preview.classList.add('preview-image');
-        preview.setAttribute('data-item-id', i);
-        preview.setAttribute('src', image.src);
-        preview.setAttribute('alt', image.alt);
-        preview.addEventListener('click', function(event) {
-            event.preventDefault();
-            var pic = lightbox.querySelector('img');
-            pic.setAttribute('src', image.src);
-            pic.setAttribute('alt', image.alt);
-            pic.setAttribute('data-item-id', i);
-            lightbox.classList.toggle('inactive');
+            var $preview = $('<img>');
+            $preview.addClass('preview-image');
+            $preview.attr('data-item-id', i);
+            $preview.attr('src', image.src);
+            $preview.attr('alt', image.alt);
+            $previewbox.append($preview);
```

Comment:
how's your load time? Do the images all load stuttered? If so, you could append all the images to a "non-drawn" element then after the loop just append the one element to the page so that they are all being loaded at the same time rather than throughout each loop, however this might be irrelevant for this many images.

Reply:
I haven't run into any issues with it yet, but that's a good idea.
Mind 👀-ing this change again?